File: LongestCommonPrefix/LonComPref.js
var longestCommonPrefix = function (strs) {
  if (!strs.length) return "";
  for (let i = 0; i <= strs[0].length; i++) {
    for (let j = 1; j < strs.length; j++) {
      if (strs[0][i] !== strs[j][i]) {
        return strs[0].slice(0, i);
      }
    }
  }
  return strs[0];
};


// ! Второе решение
var longestCommonPrefix = function (strs) {
    if (!strs.length) return '';
    for (let i = 0; i <= strs[0].length; i++) {
        if (!strs.every((string) => string[i] === strs[0][i])) {
            return strs[0].slice(0, i);
        }
    }
    return strs[0];
};

File: LongestCommonPrefix/LonComPref.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        short_str = min(strs, key=len)
        for i,x in enumerate(short_str):
            for other_str in strs:
                if other_str[i] != x:
                    return short_str[:i]
        return short_str
